### PR TITLE
Yaml Support

### PR DESCRIPTION
**Pull request by koaning**
This is a work in progress PR to add Yaml support.

There's a few things that changed. 

- I've added new features
- I'm also adding tests to check for optional dependencies. 
- I've fixed a few bugs. Some of the docstrings had usage examples that weren't properly closed off with backticks. It is now fixed.

### CODE REVIEW

**Comment by koaning** [comment]
@samarpan-rai @samukweku you don't have to, but if you feel like it -> feel free to review this.

File: clumper/clump.py [hunk 1]
```diff
@@ -131,6 +132,89 @@ def read_jsonl(cls, path: str, n=None):
...
+
+        ```python
+        from clumper import Clumper
+
+        clump = Clumper.read_yaml("tests/data/demo-flat.yml")
+        assert len(clump) == 3
+        ```
+        """
+        # Case 1 : Open cloud file in stream
+        if path.startswith("https:") or path.startswith("http:"):
```
Comment (samukweku) [inline clumper/clump.py:165]: not essential; this can be rewritten as ``path.startswith(('https:', 'http'))``

**Review by samukweku — COMMENTED** [review]
not sure if yaml has to deal with null values or empty strings.

**Comment by koaning** [comment]
Before merging this I should add a test case that checks that it can fetch from online. Then I can also test for http/https to play safe.

**Pull request merged**